modules/ingester/instance_search.go

```diff
@@ -32,11 +35,67 @@ func (i *instance) Search(ctx context.Context, req *tempopb.SearchRequest) (*tem
 		maxResults = 20
 	}
 
+	span.LogFields(ot_log.String("SearchRequest", req.String()))
 	if api.IsTraceQLQuery(req) {
-		// TODO actually implement recent traceQL search
+		// A trace can be live, in headBlock, completing blocks and complete blocks.
```

Comment:
instead of writing all this new code here to can't we just add lines in these two places:

https://github.com/grafana/tempo/blob/main/modules/ingester/instance_search.go#L172
and 
https://github.com/grafana/tempo/blob/main/modules/ingester/instance_search.go#L258

This would hook into the existing goroutine/channel/pipeline thing we already have built?

Reply:
Basically if it's traceql do something like:

```
	res, err := engine.Execute(ctx, req, traceql.NewSpansetFetcherWrapper(func(ctx context.Context, req traceql.FetchSpansRequest) (traceql.FetchSpansResponse, error) {
		return i.headBlock.Fetch(ctx, req, opts)
	}))
```

else do what it's currently doing?

Reply:
updated.